### DISCUSSION TITLE
MAX_ITER vs IMS_PER_BATCH

### DISCUSSION THREAD

**Discussion by ghazni123** [facebookresearch/detectron2#2527] [Q&A] [answered]
What is the relationship between SOLVER.MAX_ITER vs SOLVER.IMS_PER_BATCH?
Suppose
Image count in Dataset: 100 images
SOLVER.MAX_ITER: 100 iterations
SOLVER.IMS_PER_BATCH: 10 images per batch

Does one iteration in SOLVER.MAX_ITER means do training for one image? i.e. 100 iterations means 100 images (that is one epoch or pass through dataset 1 time)

OR

Does one iteration in SOLVER.MAX_ITER means do training for one batch (image)? i.e. 100 iterations means 100 batch (one batch is 10 images i.e. 1000 images). That is like 10 epochs or pass through dataset 10 times.

Could you please help me understand this? Thank you.

**Comment by ppwwyyxx** [comment, marked as answer]
> 100 iterations means 100 batch (one batch is 10 images i.e. 1000 images)

Yes.
